From cdbd0cabce1aa4cedc8089ddc462dffd9b553b48 Mon Sep 17 00:00:00 2001
From: Camber Huang <camber@poi.science>
Date: Mon, 9 May 2022 14:04:06 +0800
Subject: [PATCH] Fix build failure on riscv64 caused by missing syscall

On RISC-V 64-bit linux systems, syscall `dup2()` is not available, therefore
syscall.Dup2 will be undefined and gotop will fail to be built from soure.
This patch updates building target of logging/logging_{other,arm64} and
renames them based on syscall used.

Signed-off-by: Camber Huang <camber@aosc.io>
---
 logging/{logging_other.go => logging_dup2.go} | 2 +-
 logging/{logging_arm64.go => logging_dup3.go} | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)
 rename logging/{logging_other.go => logging_dup2.go} (67%)
 rename logging/{logging_arm64.go => logging_dup3.go} (75%)

diff --git a/logging/logging_other.go b/logging/logging_dup2.go
similarity index 67%
rename from logging/logging_other.go
rename to logging/logging_dup2.go
index d7d75d5..c7371c4 100644
--- a/logging/logging_other.go
+++ b/logging/logging_dup2.go
@@ -1,4 +1,4 @@
-// +build linux,!arm64 openbsd,!arm64 freebsd darwin
+// +build linux,!arm64,!riscv64 openbsd,!arm64 freebsd darwin
 
 package logging
 
diff --git a/logging/logging_arm64.go b/logging/logging_dup3.go
similarity index 75%
rename from logging/logging_arm64.go
rename to logging/logging_dup3.go
index 9d62f0f..966bef4 100644
--- a/logging/logging_arm64.go
+++ b/logging/logging_dup3.go
@@ -1,4 +1,4 @@
-// +build !freebsd,!darwin,arm64
+// +build !freebsd,!darwin,arm64,riscv64
 
 package logging
 
-- 
2.36.0

